htdocs/product/agenda.php

```diff
@@ -63,8 +63,10 @@
 if (! $sortfield) $sortfield='a.datep,a.id';
 if (! $sortorder) $sortorder='DESC,DESC';
 
+// Initialize hook context
+$contextpage = array('agendathirdparty','globalcard');
```

Comment:
There is a confusion with the goal of $contextpage
$contextpage is a var dedicated to save the filter selection of the page and the choice of columns set into a list when there is a list on a page. It is always a string and can't be an array.

Using it for hooks is possible as it is a global var but looks dangerous as it is not dedicated to that.
The contexthook is provided to method initHooks and can be an array.

Reply:
@eldy 
désolé je le fait en français...
le soucis c'est que si un module utilise le hook "updateSession" dans le contexte "main" (main.inc.php), si on se trouve par exemple dans un hook d'un autre contexte, par exemple "productcard", la valeur du contexthook n'est pas "productcard" mais "main" ! il faudrait corriger la manière dont le contexte est lié au hooks présents dans ce dernier.